### PR TITLE
DSDEEPB-1798 getMaximumAccessLevel traversing graph ACL nodes

### PR DESCRIPTION
**Pull request by helgridly**
- [x] **Submitter**: Rebase to develop. DO NOT SQUASH
- [x] **Submitter**: Make sure Swagger is updated if API changes
- [x] **Submitter**: Make sure documentation for code is complete
- [x] **Submitter**: Review code comments; remove done TODOs, create stories for remaining TODOs
- [x] **Submitter**: Include the JIRA issue number in the PR description
- [x] **Submitter**: Add description or comments on the PR explaining the hows/whys (if not obvious)
- [x] Tell  that the PR exists if he wants to look at it
- [x] Anoint a lead reviewer (LR). **Assign PR to LR**
- [x] **LR**: Initial review by LR and others.
- [x] Comment / review / update cycle:
  - Rest of team may comments on PR at will
  - **LR assigns to submitter** for feedback fixes
  - Submitter updates documentation as needed
  - Submitter rebases to develop again if necessary
  - Submitter makes further commits. DO NOT SQUASH. **Reassign to LR** for further feedback
- [x]  sign off
- [x] **LR** sign off
- [x] **Assign to submitter** to finalize
- [x] **Submitter**: Squash commits, rebase if necessary
- [x] **Submitter**: Verify all tests go green, including CI tests
- [x] **Submitter**: Merge to develop 
- [x] **Submitter**: Delete branch after merge
- [x] **Submitter**: Check configuration files in Jenkins in case they need changes
- [x] **Submitter**: Verify swagger UI on dev server still works after deployment
- [x] **Submitter**: Inform other teams of any API changes via hipchat and/or email
- [x] **Submitter**: Mark JIRA issue as resolved once this checklist is completed

### CODE REVIEW

**Comment by helgridly** [comment]
@dvoet This PR adds a graph-y implementation of `getMaximumAccessLevel`.

**Comment by helgridly** [comment]
The history is a bit weird here because I did a rebase. I'll fix it on Monday.

(note to self: removing the duplicated lines in joel's fix is what caused this, i think.)

File: src/main/scala/org/broadinstitute/dsde/rawls/dataaccess/GraphAuthDAO.scala [hunk 1]
```diff
@@ -43,4 +51,49 @@ class GraphAuthDAO extends AuthDAO with GraphDAO {
...
+  override def getMaximumAccessLevel(userSubjectId: String, workspaceId: String, txn: RawlsTransaction): WorkspaceAccessLevel = {
+    txn withGraph { db =>
+
+      //Returns a list of paths starting with a group the user belongs in and ending with the workspace's access map.
+      // user -> group -> ...some other groups... -> group -> workspace access map
+      //Multiple paths implies that the user belongs to multiple groups associated with the workspace.
+      val accessMapPaths =
+        userPipeline(db, userSubjectId).as("foo").in()
+          .loop(
+            "foo", //start point of loop
```
Comment (dvoet) [inline src/main/scala/org/broadinstitute/dsde/rawls/dataaccess/GraphAuthDAO.scala]: perhaps a more descriptive name that foo, may user?

**Comment by dvoet** [comment]
fix the foo then :+1: from me

File: src/main/scala/org/broadinstitute/dsde/rawls/dataaccess/GraphAuthDAO.scala (hunk 1 above)
Comment (helgridly) [inline src/main/scala/org/broadinstitute/dsde/rawls/dataaccess/GraphAuthDAO.scala]: The user isn't always the "current" node in the loop condition, but I'll make it a different word regardless. Maybe bar.

**Comment by helgridly** [comment]
@jmthibault79 did you :+1: this?

**Comment by jmthibault79** [comment]
No, I have not yet reviewed this.

**Comment by helgridly** [comment]
Oh, sorry for squashing then :/

**Comment by jmthibault79** [comment]
Easier to review when squashed anyway.  Now I have a :+1: for you.

**Pull request merged**